### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by Wahaj-Ali**
## Highlights
- Limit access to web app resources based on authorization rules.
- Install CanCanCan.
- Added a role column to the users' table. Used a migration for this.
- Only authorized users can see the Delete button for posts and comments.
- A user can delete only his posts and comments.
- An admin can delete all posts and comments

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -28,4 +28,18 @@ def create
       render :new
     end
   end
+
```
Comment (Ismailco) [inline app/controllers/posts_controller.rb:33]: - [x] You should use `CanCanCan` for Post authorization in `Post Controller`, kindly make sure to implement this. you can add this line to your controller ``load_and_authorize_resource``

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Wahaj-Ali, @codehass, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Wahaj-Ali) [reply app/controllers/posts_controller.rb:33]: Thank you @Ismailco for pointing out the issue. I have resolved it 😄

**Review by Owusu-Desmond — APPROVED** [review]
Hi @Wahaj-Ali and @Ismailco , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by codehass** [comment]
Hi @Owusu-Desmond thank you for your review, you approved the project in my dashboard can you cancel it, we should submit it individually.
Thank you 😁

**Pull request merged**